### ISSUE TITLE
"jsdoc/no-undefined-types" crashes in files with links to unknown types

### ISSUE THREAD

**Issue by milang** [gajus/eslint-plugin-jsdoc#1079]
## Expected behavior

ESLint should not crash while linting a file with JSDoc that contains link to an unknown type. An ESLint issue should be reported instead.

## Actual behavior

ESLint crashes when processing `jsdoc/no-undefined-types` rule in a file that contains link to an unknown type.

Example problematic JSDoc:

```tsx
/**
 * Message with {@link NotKnown}
 */
```

Example crash stack trace:

```text
TypeError: Cannot read properties of undefined (reading '0')
Occurred while linting /home/projects/jsdoc-no-undefined-types-issue/index.js:4
Rule: "jsdoc/no-undefined-types"
    at report (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint-plugin-jsdoc@44.2.2_eslint@8.40.0/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js:782:40)
    at eval (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint-plugin-jsdoc@44.2.2_eslint@8.40.0/node_modules/eslint-plugin-jsdoc/dist/rules/noUndefinedTypes.js:168:13)
    at _traverse (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/jsdoc-type-pratt-parser@4.0.0/node_modules/jsdoc-type-pratt-parser/dist/index.js:2500:59)
    at traverse (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/jsdoc-type-pratt-parser@4.0.0/node_modules/jsdoc-type-pratt-parser/dist/index.js:2524:9)
    at _default.iterateAllJsdocs (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint-plugin-jsdoc@44.2.2_eslint@8.40.0/node_modules/eslint-plugin-jsdoc/dist/rules/noUndefinedTypes.js:159:32)
    at iterate (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint-plugin-jsdoc@44.2.2_eslint@8.40.0/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js:852:3)
    at callIterator (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint-plugin-jsdoc@44.2.2_eslint@8.40.0/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js:953:7)
    at *:not(Program) (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint-plugin-jsdoc@44.2.2_eslint@8.40.0/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js:998:11)
    at ruleErrorHandler (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint@8.40.0/node_modules/eslint/lib/linter/linter.js:1049:28)
    at eval (file:///home/projects/jsdoc-no-undefined-types-issue/node_modules/.pnpm/eslint@8.40.0/node_modules/eslint/lib/linter/safe-emitter.js:45:58)
```

## ESLint Config

Minimal project that reproduces the issue:
https://stackblitz.com/edit/jsdoc-no-undefined-types-issue?file=index.js&view=editor

(StackBlitz should run linting automatically after opening the above URL, run `pnpm run dev` in the terminal if it doesn't)

## Environment

- Node version: `16.14.2`
- ESLint version: `8.40.0`
- `eslint-plugin-jsdoc` version: `44.2.2` (there was no crash in `44.1.0`)

**Comment by milang** [comment]
Thank you! I can confirm the issue has been fixed in `44.2.3`.
